lib/private/Share20/Manager.php

```diff
@@ -1784,9 +1784,21 @@ public function shareApiAllowLinks() {
 	/**
 	 * Is password on public link requires
 	 *
+	 * @param bool Check group membership exclusion
 	 * @return bool
 	 */
-	public function shareApiLinkEnforcePassword() {
+	public function shareApiLinkEnforcePassword(bool $checkGroupMembership = true) {
+		$excludedGroups = $this->config->getAppValue('core', 'shareapi_enforce_links_password_excluded_groups', '');
+		if ($excludedGroups !== '' && $checkGroupMembership) {
+			$excludedGroups = json_decode($excludedGroups);
+			$user = $this->userSession->getUser();
+			if ($user) {
+				$userGroups = $this->groupManager->getUserGroupIds($user);
+				if ((bool)array_intersect($excludedGroups, $userGroups)) {
+					return false;
+				}
+			}
+		}
```

Comment:
More explicit conditions and minor refactor?

```suggestion
		$user = $this->userSession->getUser();
		if ($checkGroupMembership && $user !== null) {
			$userGroups = $this->groupManager->getUserGroupIds($user);
			$excludedGroups = json_decode($this->config->getAppValue('core', 'shareapi_enforce_links_password_excluded_groups', '[]'));
			if (count(array_intersect($excludedGroups, $userGroups)) > 0) {
				return false;
			}
		}
```